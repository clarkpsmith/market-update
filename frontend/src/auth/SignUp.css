.SignUp {
  text-align: left;
  color: white;
  display: flex;
  justify-content: center;
  background-color: black;
  height: 100vh;
  border-radius: 1px;
}

.SignUp-form {
  text-align: left;
  padding: 25px;
  padding-left: 50px;
  width: 400px;
  background-color: black;
  border: 1px solid grey;
  border-radius: 2px;
}

.SignUp-input {
  border-radius: 3px;

  width: 90%;
  border: 1px solid rgb(216, 213, 213);
}

.SignUp-btn {
  margin: 25px;
  float: right;
  margin-right: 35px;
}

.SignUp-block {
  margin: 50px;
}

.SignUp-title {
  padding: 10px;
  text-align: center;
  color: rgb(200, 210, 236);
}

.delete-profile {
  text-align: right;
  padding-right: 0px;
  background-color: black;
  padding-bottom: 200px;
}

@media only screen and (max-width: 414px) {
  .delete-profile {
    padding-right: 60px;
  }
}
