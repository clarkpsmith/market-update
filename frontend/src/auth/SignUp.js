import React, { useState } from "react";
import "../common/SearchForm.css";
import { Button, Form, FormGroup } from "reactstrap";
import { useHistory } from "react-router-dom";
import "./SignUp.css";
import Alert from "../helpers/Alert";
import ChaseLoading from "../chaseloading/ChaseLoading";

const SignUp = ({ signUp }) => {
  const initialData = {
    username: "",
    password: "",
    firstName: "",
    lastName: "",
    email: "",
  };

  const [formData, setFormData] = useState(initialData);
  const [formErrors, setFormErrors] = useState(null);
  const [loading, setLoading] = useState(false);

  const history = useHistory();
  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData((data) => ({ ...data, [name]: value }));
  }

  const handleSubmit = async (evt) => {
    evt.preventDefault();
    setLoading(true);
    let result = await signUp(formData);
    if (result.success) {
      history.push("/watchlist");
    } else {
      setFormErrors(() => result.err);
      setLoading(false);
    }
  };
  if (loading) {
    return (
      <div className="Login">
        <ChaseLoading />
      </div>
    );
  }
  return (
    <div className="SignUp">
      <div className="SignUp-block">
        <h4 className="SignUp-info">
          Want to create a personal watchlist of your favorite stocks?
        </h4>
        <h3 className="SignUp-title">Sign Up</h3>
        <Form className="SignUp-form" onSubmit={handleSubmit}>
          <FormGroup>
            <label htmlFor="username">Username</label>
            <br />
            <input
              className="SignUp-input"
              type="text"
              name="username"
              id="username"
              value={FormData.username}
              onChange={handleChange}
              required
            />
            <br />
          </FormGroup>
          <FormGroup>
            <label htmlFor="password">Password</label>
            <br />
            <input
              className="SignUp-input"
              type="password"
              name="password"
              id="password"
              value={FormData.password}
              onChange={handleChange}
              required
            />
            <br />
          </FormGroup>
          <FormGroup>
            <label htmlFor="firstName">First Name</label>
            <br />
            <input
              className="SignUp-input"
              type="text"
              name="firstName"
              id="firstName"
              value={FormData.firstName}
              onChange={handleChange}
              required
            />
          </FormGroup>

          <FormGroup>
            <label htmlFor="lastName">Last Name</label>
            <br />
            <input
              className="SignUp-input"
              type="text"
              name="lastName"
              id="lastName"
              value={FormData.lastName}
              onChange={handleChange}
              required
            />
          </FormGroup>
          <FormGroup>
            <label htmlFor="email">Email</label>
            <br />
            <input
              className="SignUp-input"
              type="text"
              name="email"
              id="email"
              value={FormData.email}
              onChange={handleChange}
              required
            />
          </FormGroup>
          {formErrors ? <Alert type="danger" messages={formErrors} /> : null}
          <Button size="md" className="SignUp-btn">
            Submit
          </Button>
          <br />
          <br />
        </Form>
      </div>
    </div>
  );
};
export default SignUp;
