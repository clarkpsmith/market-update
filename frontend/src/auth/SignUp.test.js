import React from "react";
import { render } from "@testing-library/react";
import SignUp from "./SignUp";

it("renders without crashing", () => {
  render(<SignUp />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<SignUp />);
  expect(asFragment()).toMatchSnapshot();
});
