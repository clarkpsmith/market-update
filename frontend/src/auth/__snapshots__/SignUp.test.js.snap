// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SignUp"
  >
    <div
      class="SignUp-block"
    >
      <h3
        class="SignUp-title"
      >
        Sign Up
      </h3>
      <form
        class="SignUp-form"
      >
        <div
          class="form-group"
        >
          <label
            for="username"
          >
            Username
          </label>
          <br />
          <input
            class="SignUp-input"
            id="username"
            name="username"
            required=""
            type="text"
            value=""
          />
          <br />
        </div>
        <div
          class="form-group"
        >
          <label
            for="password"
          >
            Password
          </label>
          <br />
          <input
            class="SignUp-input"
            id="password"
            name="password"
            required=""
            type="password"
            value=""
          />
          <br />
        </div>
        <div
          class="form-group"
        >
          <label
            for="firstName"
          >
            First Name
          </label>
          <br />
          <input
            class="SignUp-input"
            id="firstName"
            name="firstName"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="lastName"
          >
            Last Name
          </label>
          <br />
          <input
            class="SignUp-input"
            id="lastName"
            name="lastName"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="email"
          >
            Email
          </label>
          <br />
          <input
            class="SignUp-input"
            id="email"
            name="email"
            required=""
            type="text"
            value=""
          />
        </div>
        <button
          class="SignUp-btn btn btn-secondary btn-md"
        >
          Submit
        </button>
        <br />
        <br />
      </form>
    </div>
  </div>
</DocumentFragment>
`;
